src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1092,7 +1096,22 @@ private void EmitSwitchStatement(BoundSwitchStatement switchStatement)
                         var value = boundLabel.ConstantValueOpt;
                         Debug.Assert(value != null
                             && SwitchConstantValueHelper.IsValidSwitchCaseLabelConstant(value));
-                        labelsBuilder.Add(new KeyValuePair<ConstantValue, object>(value, label));
+
+                        // in debug we should be able to set breakpoints on individual case lables, 
```

Comment:
We probably should. We do allow breakpoints on regular labels. That could be useful during debugging - to catch when a particular "case" was actually hit. 

But you are right. We don't. 
I am still not sure if there is any value in deoptimizing this in Debug case or not.